catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportBuilderTest.java

```diff
@@ -0,0 +1,455 @@
+        Document htmlDoc = xmlFileToDocument(html);
+        assertSummaryReportValues(htmlDoc);
+    }
+
+    @Test
+    public void testSummaryReportWithSparseRegistryPackage() throws Exception {
+        MetacardImpl metacard = convertXmltoMetacard("/csw-sparse-registry-package.xml");
+        String html = reportHelper.getSummaryHtmlFromMetacard(metacard);
+        Document htmlDoc = xmlFileToDocument(html);
+        assertSparseSummaryReportValues(htmlDoc);
+    }
+
+    @Test
+    public void testSummaryReportWithBareRegistryPackage() throws Exception {
+        MetacardImpl metacard = convertXmltoMetacard("/csw-bare-registry-package.xml");
+        String html = reportHelper.getSummaryHtmlFromMetacard(metacard);
+        Document htmlDoc = xmlFileToDocument(html);
+        String organizationTitle = htmlDoc.getElementsByTagName("h2")
+                .item(0)
+                .getTextContent();
+        assertThat(organizationTitle, is("Summary"));
+        assertThat(htmlDoc.getElementsByTagName("table")
+                .item(0)
+                .getTextContent(), is("            "));
+    }
+
+    @Test
+    public void testOrganizationReportWithSimplifiedRegistryPackage() throws Exception {
+        RegistryPackageType registryPackage = getRegistryPackageFromResource(
+                "/csw-simplified-registry-package.xml");
+        String html = reportHelper.getHtmlFromRegistryPackage(registryPackage,
+                reportHelper.ORGANIZATIONS);
+        Document htmlDoc = xmlFileToDocument(html);
+        assertOrganizationReportValues(htmlDoc);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testOrganizationReportWithBareRegistryPackage() throws Exception {
+        RegistryPackageType registryPackage = getRegistryPackageFromResource(
+                "/csw-bare-registry-package.xml");
+        String html = reportHelper.getHtmlFromRegistryPackage(registryPackage,
+                reportHelper.ORGANIZATIONS);
+        Document htmlDoc = xmlFileToDocument(html);
+
+        String organizationTitle = htmlDoc.getElementsByTagName("h2")
+                .item(0)
+                .getTextContent();
+        assertThat(organizationTitle, is("Organizations"));
+
+        htmlDoc.getElementsByTagName("table")
+                .item(0)
+                .getTextContent();
+    }
+
+    @Test
+    public void testFullReportWithSimplifiedRegistryPackage() throws Exception {
+        RegistryPackageType registryPackage = getRegistryPackageFromResource(
+                "/csw-simplified-registry-package.xml");
+        String html = reportHelper.getHtmlFromRegistryPackage(registryPackage, reportHelper.REPORT);
+        Document htmlDoc = xmlFileToDocument(html);
+        assertFullReportValues(htmlDoc);
+    }
+
+    private void assertSummaryReportValues(Document htmlDoc) {
+        String organizationName = htmlDoc.getElementsByTagName("strong")
```

Comment:
Not sure if I can think of a great way around this, but any concerns about the report changing fields, values, and having to re-writing a lot of the testing as a result?